### PR TITLE
Remove the experimental Utf8String type and tests from the runtime repo

### PR DESCRIPTION
**Pull request by trylek**
I'm proposing to remove the Utf8String and Utf8Span types and the
System.Utf8String.Experimental tests from the runtime repo as according to
JanK's clarifications these types and tests are currently treated as experimental
and belong to the runtimelab, not to the "production" runtime repo. From a
practical perspective, we're hitting crashes in the Utf8String tests when using
System.Private.CoreLib compiled with Crossgen2 which doesn't yet implement
the Utf8String-related typesystem logic.

Thanks

Tomas

Related to: https://github.com/dotnet/runtime/issues/41521

### CODE REVIEW

File: src/coreclr/clr.featuredefines.props [hunk 1]
```diff
@@ -38,7 +38,6 @@
 
     <!-- Features we're currently flighting, but don't intend to ship in officially supported releases -->
```
Comment (jkotas) [inline src/coreclr/clr.featuredefines.props:39]: You can delete this. We have runtimelab for this now.

File: src/coreclr/clrdefinitions.cmake [hunk 2]
```diff
@@ -2,7 +2,6 @@ include(clrfeatures.cmake)
 
 # Features we're currently flighting, but don't intend to ship in officially supported releases
```
Comment (jkotas) [inline src/coreclr/clrdefinitions.cmake:3]: Dtto

File: src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 3]
```diff
@@ -253,9 +253,6 @@
     <Compile Include="$(BclSourcesRoot)\Internal\Runtime\InteropServices\ComActivator.cs" />
     <Compile Include="$(BclSourcesRoot)\Internal\Runtime\InteropServices\InMemoryAssemblyLoader.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(FeatureUtf8String)' == 'true'">
```
Comment (jkotas) [inline src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj:256]: There is also a block in `System.Private.CoreLib.Shared.projitems` that should be deleted

**Comment by GrabYourPitchforks** [comment]
Don't forget to remove the file __runtime/src/libraries/System.Private.CoreLib/src/System/Char8.cs__ and all references to the __Char8__ type as well.

File: src/coreclr/clr.featuredefines.props (hunk 1 above)
Comment (trylek) [reply src/coreclr/clr.featuredefines.props:39]: Good point, removed in 3rd commit.

File: src/coreclr/clrdefinitions.cmake (hunk 2 above)
Comment (trylek) [reply src/coreclr/clrdefinitions.cmake:3]: Removed in 3rd commit.

File: src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj (hunk 3 above)
Comment (trylek) [reply src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj:256]: Thanks for pointing that out, removed in 3rd commit.

**Review by jkotas — APPROVED** [review]

**Review by GrabYourPitchforks — APPROVED** [review]

**Comment by trylek** [comment]
Merging now to unblock my follow-up Crossgen2 efforts. I'll be happy to address any additional feedback in a follow-up PR.

**Pull request merged**